src/coreclr/gc/gc.cpp

```diff
@@ -45589,6 +45589,18 @@ void PopulateDacVars(GcDacVars *gcDacVars)
     };
     static_assert(sizeof(gc_heap_field_offsets) == (GENERATION_TABLE_FIELD_INDEX + 1) * sizeof(int), "GENERATION_TABLE_INDEX mismatch");
 #endif //MULTIPLE_HEAPS
+    static int generation_field_offsets[] = {
+
+#define DEFINE_FIELD(field_name, field_type) offsetof(generation, field_name),
+#define DEFINE_DPTR_FIELD(field_name, field_type) offsetof(generation, field_name),
+#define DEFINE_MISSING_FIELD -1,
+
+#include "generation_fields.h"
+
+#undef DEFINE_MISSING_FIELD
+#undef DEFINE_DPTR_FIELD
+#undef DEFINE_FIELD
```

Comment:
do you really need to define and undefine all these macros for each type? could you define a macro for the class name so you don't need to repeat this? eg

```cplusplus
#define DEFINE_FIELD(field_name, field_type) offsetof(CLASS_NAME, field_name),
#define DEFINE_MISSING_FIELD -1,

static int gc_heap_field_offsets[] = {
#define CLASS_NAME gc_heap
#include "gcheap_fields.h"
};
```